### PR TITLE
add cols kwarg to rename/rename!

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/3361

### CODE REVIEW

**Comment by bkamins** [comment]
Having said that, I think it is OK to add more functionalities to meta-packages (like DataFramesMeta.jl or Tidier.jl) if they would be indeed more useful.

CC @pdeffebach @kdpsingh

File: src/abstractdataframe/abstractdataframe.jl [hunk 1]
```diff
@@ -132,8 +132,10 @@ Each name is changed at most once. Permutation of names is allowed.
 - `df` : the `AbstractDataFrame`
 - `d` : an `AbstractDict` or an `AbstractVector` of `Pair`s that maps
   the original names or column numbers to new names
-- `f` : a function which for each column takes the old name as a `String`
-  and returns the new name that gets converted to a `Symbol`
+- `f` : a function which for each column selected by `cols` keyword argument
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:135]:
```suggestion
- `f` : a function which for each column selected by the `cols` keyword argument
```

File: src/abstractdataframe/abstractdataframe.jl [hunk 2]
```diff
@@ -277,8 +283,10 @@ Each name is changed at most once. Permutation of names is allowed.
   only allowed if it was created using `:` as a column selector.
 - `d` : an `AbstractDict` or an `AbstractVector` of `Pair`s that maps
   the original names or column numbers to new names
-- `f` : a function which for each column takes the old name as a `String`
-  and returns the new name that gets converted to a `Symbol`
+- `f` : a function which for each column selected by `cols` keyword argument
```
Comment (nalimilan) [inline src/abstractdataframe/abstractdataframe.jl:286]:
```suggestion
- `f` : a function which for each column selected by the `cols` keyword argument
```

**Review by nalimilan — APPROVED** [review]

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**